lib/rexml/parsers/baseparser.rb

```diff
@@ -318,7 +318,11 @@ def pull_event
               raise REXML::ParseException.new( "Bad ELEMENT declaration!", @source ) if md.nil?
               return [ :elementdecl, "<!ELEMENT" + md[1] ]
             elsif @source.match("ENTITY", true)
-              match = [:entitydecl, *@source.match(Private::ENTITYDECL_PATTERN, true, term: Private::ENTITY_TERM).captures.compact]
+              scanner = @source.match(Private::ENTITYDECL_PATTERN, true, term: Private::ENTITY_TERM)
```

Comment:
Could you use `match_data` not `scanner`?
We want to use `MatchData` instead of `StringScanner` because we don't want to use `StringScanner` in `REXML::Source` directly. But `REXML::Source#match` returns `StringScanner` directly for performance reason.
So we should assume that the return value of `REXML::Source#match` is a `MatchData` and we should only use `MatchData` compatible API provided by `StringScanner`.

Reply:
I see, so that's the intention.
https://github.com/ruby/rexml/pull/182/commits/563bb5737d0e8420d70e78d25f330327bfde5a9a